libs/execution_context/ebpf_program.c

```diff
@@ -107,12 +114,14 @@ _ebpf_program_program_info_provider_changed(
     ebpf_result_t return_value;
     ebpf_program_t* program = (ebpf_program_t*)client_binding_context;
     uint32_t* provider_helper_function_ids = NULL;
+    uint32_t* global_helper_function_ids = NULL;
 
     if (provider_data == NULL) {
         // Extension is detaching. Program will get invalidated.
         goto Exit;
     } else {
         ebpf_helper_function_addresses_t* helper_function_addresses = NULL;
```

Comment:
```suggestion
        ebpf_helper_function_addresses_t* program_type_specific_helper_function_addresses = NULL;
```

Reply:
Changed to `program_specific_helper_function_addresses`. Though this has caused changes in many more files in the PR.